bloom-filter/src/main/scala/bloomfilter/mutable/CuckooFilter.scala

```diff
@@ -64,7 +65,7 @@ object CuckooFilter {
   }
 
   def optimalNumberOfBuckets(numberOfItems: Long): Long = {
-    var numberOfBuckets = upperPowerOf2(numberOfItems / UnsafeTable16Bit.TagsPerBucket)
+    var numberOfBuckets = upperPowerOf2((numberOfItems + UnsafeTable16Bit.TagsPerBucket - 1) / UnsafeTable16Bit.TagsPerBucket)
```

Comment:
You're right!

Initially I thought that the original reference calculation is the right one.

Here's an example:
We expect to put 1026 items. In case of the origin version `upperPowerOf2(numberOfItems / UnsafeTable16Bit.TagsPerBucket)` we get `upperPowerOf2(1026/4) = upperPowerOf2(256) = 256` which doesn't cover all items.
In your version `upperPowerOf2((numberOfItems + UnsafeTable16Bit.TagsPerBucket - 1) / UnsafeTable16Bit.TagsPerBucket)` we have `upperPowerOf2((1026 + 4 - 1)/4) = upperPowerOf2(257) = 512` which is the right number. 
There's the fraction/filling check I borrowed from the original version https://github.com/efficient/cuckoofilter/blob/8b573bf804ca067c08f973c0e8ef18b5d8e8768e/src/cuckoofilter.h#L89 which hides that miscalculation.